common-items.lic

```diff
@@ -753,9 +753,19 @@ module DRCI
       return nil unless open_container?(container)
       rummage_container(container)
     else
-      contents
-        .match(/You rummage through .* and see (?:a|an|some) (?<items>.*)\./)[:items]
-        .split(/(?:, | and )?(?:a|an|some) /)
+      # Get string of just the comma separated item list
+      contents = contents.match(/You rummage through .* and see (?:a|an|some) (?<items>.*)\./)[:items]      
+      
+      # Split at a, an, or some, but only when it follows a comma
+      # The last two items in the rummage list are never comma delimited, handle shortly
+      contents = contents.split(/, (?:a|an|some) /)      
+      
+      # Grab the last array element which is the final two items that are never comma
+      # delimited in game output, split them into their item descriptions
+      sub_contents = contents.last.split(/ and (?:a|an|some) /)
```

Comment:
We also need to remove the last element from the list otherwise the last two items are repeated as ["x and y", "x", "y"]

Change out `.last` with `.pop` to get _and_ remove that last item that you then split then tack on in the next line.
```suggestion
      sub_contents = contents.pop.split(/ and (?:a|an|some) /)
```

Reply:
Ah, great catch. I hadn't even noticed that. Thank you.